modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -598,7 +601,8 @@ private CompletableFuture<?> createZonePartitionReplicationNode(
             Assignments stableAssignments,
             long revision,
             int partitionCount,
-            boolean isVolatileZone
+            boolean isVolatileZone,
+            boolean holdingZoneWriteLock
```

Comment:
Why don't we always call this method under lock instead?

Reply:
If we do it, then pending assignments change events handling (inside the same zone) will be sequential. Imagine that partitions 1, 2, 3 of the same zone are now added to pending assignments of node A. If we just take write lock for any of them, others will need to wait for it. But current approach allows to keep this parallelism

Reply:
Also, when a new zone is created, its partitions would be started sequentially if we took a write lock for each of them